crates/hyperdrive-math/src/long/targeted.rs

```diff
@@ -0,0 +1,518 @@
+    /// $$
+    ///
+    /// And finally, the price after long derivative is:
+    ///
+    /// $$
+    /// p'(x) = v'(x) T v(x)^(T-1)
+    /// $$
+    ///
+    fn price_after_long_derivative(
+        &self,
+        base_amount: FixedPoint,
+        bond_amount: FixedPoint,
+    ) -> Option<FixedPoint> {
+        // g'(x)
+        let gov_fee_derivative =
+            self.governance_lp_fee() * self.curve_fee() * (fixed!(1e18) - self.get_spot_price());
+
+        // a(x) = u (z_e + x/c - g(x) - zeta)
+        let inner_numerator = self.mu()
+            * (self.ze() + base_amount / self.vault_share_price()
+                - self.open_long_governance_fee(base_amount)
+                - self.zeta().into());
+
+        // a'(x) = u / c - g'(x)
+        let inner_numerator_derivative = self.mu() / self.vault_share_price() - gov_fee_derivative;
+
+        // b(x) = y_0 - Y(x)
+        let inner_denominator = self.bond_reserves() - bond_amount;
+
+        // b'(x) = Y'(x)
+        let long_amount_derivative = match self.long_amount_derivative(base_amount) {
+            Some(derivative) => derivative,
+            None => return None,
+        };
+
+        // v(x) = a(x) / b(x)
+        // v'(x) = ( b(x) * a'(x) + a(x) * b'(x) ) / b(x)^2
+        let inner_derivative = (inner_denominator * inner_numerator_derivative
+            + inner_numerator * long_amount_derivative)
+            / inner_denominator.pow(fixed!(2e18));
+
+        // p'(x) = v'(x) T v(x)^(T-1)
+        // p'(x) = v'(x) T v(x)^(-1)^(1-T)
+        // v(x) is flipped to (denominator / numerator) to avoid a negative exponent
+        return Some(
+            inner_derivative
+                * self.time_stretch()
+                * (inner_denominator / inner_numerator).pow(fixed!(1e18) - self.time_stretch()),
+        );
+    }
+
+    /// Calculate the base & bond deltas from the current state given desired new reserve levels.
+    ///
+    /// Given a target ending pool share reserves, $z_t$, and bond reserves, $y_t$,
+    /// the trade deltas to achieve that state would be:
+    ///
+    /// $$
+    /// \Delta x = c * (z_t - z_e) \\
+    /// \Delta y = y - y_t - c(\Delta x)
+    /// $$
+    ///
+    /// where $c$ is the vault share price and
+    /// $c(\Delta x)$ is the (open_long_curve_fee)[long::fees::open_long_curve_fees].
+    fn trade_deltas_from_reserves(
+        &self,
+        share_reserves: FixedPoint,
+        bond_reserves: FixedPoint,
+    ) -> (FixedPoint, FixedPoint) {
+        let base_delta =
+            (share_reserves - self.effective_share_reserves()) * self.vault_share_price();
+        let bond_delta =
+            (self.bond_reserves() - bond_reserves) - self.open_long_curve_fees(base_delta);
+        (base_delta, bond_delta)
+    }
+
+    /// Calculates the long that should be opened to hit a target interest rate.
+    /// This calculation does not take Hyperdrive's solvency constraints or exposure
+    /// into account and shouldn't be used directly.
```

Comment:
In addition to not taking Hyperdrive's solvency constraints and exposure into account, it also doesn't consider fees. With this in mind, this could be a source of our error. I haven't looked at the tests yet, but on a first glance, this seems like it will cause problems.

As we spoke about during the initial call about the targeted long whitepaper, we can't actually solve directly for this in the same way that we can solve directly for the max long and max short. Let me know if I'm missing something, but it seems more reasonable to me to use this as a guess and proceed directly to Newton's method.

Reply:
> it also doesn't consider fees.

I've been confused about this, so I might have something wrong here. However, your comment brought to my attention that the docstring itself is inaccurate. This function returns the reserve levels (shares & bonds) for a pool to hit the target rate. I believe it is accurate and would not need to account for fees.

That being said, the output of this function is immediately passed into `trade_deltas_from_reserves`, which converts the values into base & bond _deltas_. These delta amounts do take into account fees, but I suspect the two things must be considered jointly to truly get the long amount? I'd appreciate any guidance for how to communicate this.

> it seems more reasonable to me to use this as a guess and proceed directly to Newton's method.

That's what we are doing -- we take this guess and if it is within the error tolerance then we return, else we use it as the starting point for Newton's method.

Reply:
Resolving this since it's outdated. Consider my new review comments to be my current thoughts.